src/main/java/net/dv8tion/jda/core/entities/impl/ReceivedMessage.java

```diff
@@ -149,6 +149,66 @@ else if (reaction.isSelf())
         return channel.addReactionById(getIdLong(), unicode);
     }
 
+    @Override
+    public RestAction<Void> removeReaction(Emote emote)
+    {
+        Checks.notNull(emote, "Emote");
+
+        MessageReaction reaction = reactions.parallelStream()
+            .filter(r -> Objects.equals(r.getReactionEmote().getId(), emote.getId()))
+            .findFirst().orElse(null);
+
+        if (reaction == null || !reaction.isSelf())
+            return new RestAction.EmptyRestAction<>(getJDA(), null);
```

Comment:
you should throw an exception rather than pretending that the restaction was successful

Reply:
addReaction does the same thing

Reply:
Throwing an exception is not needed